pkg/dao/external_resource.go

```diff
@@ -46,3 +80,17 @@ func UrlToRepomdUrl(urlIn string) (string, error) {
 	u.Path = path.Join(u.Path, "/repodata/repomd.xml")
 	return u.String(), nil
 }
+
+func validateGpgKeyString(bodyString string) error {
+	httpError := echo.NewHTTPError(http.StatusNotAcceptable, "Received response is not a valid gpg-key")
+	if len(bodyString) < 36 {
+		return httpError
+	}
+	firstLine := strings.ToLower(bodyString[0:36])
+
+	if strings.Contains(firstLine, "begin pgp public key block") {
+		return nil
+	}
```

Comment:
should we actuall try to parse it?

Reply:
Can you point to an example of what you mean?

Reply:
i guess i mean using https://github.com/ProtonMail/gopenpgp to parse the key and ensure at least one valid key is present: https://gist.github.com/jvehent/9994311#file-pgpkeydetail-go-L17

https://pkg.go.dev/golang.org/x/crypto/openpgp#ReadArmoredKeyRing

Reply:
actually sorry, i think you'd use  ReadKeyRing maybe for an arbitrary string of keys?

Reply:
actually i'm not sure which we need, i'm dealing with this now on my task, so i'll let you know what i figure out

Reply:
The reason I went the route I did, was just to do a quick basic check on the string to make sure it had the correct syntax. If there is an equivalent of  .checkValid(publicKey) :boolean I'd love that.

Reply:
@jlsherrill, any update on this?

Reply:
its as simple as this:  `openpgp.ReadArmoredKeyRing(strings.NewReader(gpgKey))`  where gpgKey is a string with the gpg key.  Here's some sample usage:
```go
	gpgKey := "-----BEGIN PGP PUBLIC KEY BLOCK-----\nVersion: GnuPG v1.4.7 (GNU/Linux)\n\nmQENBFYxWIwBCADAKoZhZlJxGNGWzqV+1OG1xiQeoowKhssGAKvd+buXCGISZJwT\nLXZqIcIiLP7pqdcZWtE9bSc7yBY2MalDp9Liu0KekywQ6VVX1T72NPf5Ev6x6DLV\n7aVWsCzUAF+eb7DC9fPuFLEdxmOEYoPjzrQ7cCnSV4JQxAqhU4T6OjbvRazGl3ag\nOeizPXmRljMtUUttHQZnRhtlzkmwIrUivbfFPD+fEoHJ1+uIdfOzZX8/oKHKLe2j\nH632kvsNzJFlROVvGLYAk2WRcLu+RjjggixhwiB+Mu/A8Tf4V6b+YppS44q8EvVr\nM+QvY7LNSOffSO6Slsy9oisGTdfE39nC7pVRABEBAAG0N01pY3Jvc29mdCAoUmVs\nZWFzZSBzaWduaW5nKSA8Z3Bnc2VjdXJpdHlAbWljcm9zb2Z0LmNvbT6JATUEEwEC\nAB8FAlYxWIwCGwMGCwkIBwMCBBUCCAMDFgIBAh4BAheAAAoJEOs+lK2+EinPGpsH\n/32vKy29Hg51H9dfFJMx0/a/F+5vKeCeVqimvyTM04C+XENNuSbYZ3eRPHGHFLqe\nMNGxsfb7C7ZxEeW7J/vSzRgHxm7ZvESisUYRFq2sgkJ+HFERNrqfci45bdhmrUsy\n7SWw9ybxdFOkuQoyKD3tBmiGfONQMlBaOMWdAsic965rvJsd5zYaZZFI1UwTkFXV\nKJt3bp3Ngn1vEYXwijGTa+FXz6GLHueJwF0I7ug34DgUkAFvAs8Hacr2DRYxL5RJ\nXdNgj4Jd2/g6T9InmWT0hASljur+dJnzNiNCkbn9KbX7J/qK1IbR8y560yRmFsU+\nNdCFTW7wY0Fb1fWJ+/KTsC4=\n=J6gs\n-----END PGP PUBLIC KEY BLOCK-----\n"

	keyring, err := openpgp.ReadArmoredKeyRing(strings.NewReader(gpgKey))
	if err != nil {
		return err  //Bad key
	}
       //good key
```

Reply:
openpgp is deprecated, I used another package, but similar fix. Cheers